### PR TITLE
US008: Apply Repository Design Pattern F02

### PR DESCRIPTION
**Pull request by iamurtazo**

### CODE REVIEW

File: ParkingZoneApp/Areas/Admin/ParkingZoneController.cs [hunk 1]
```diff
@@ -98,8 +97,8 @@ public async Task<IActionResult> Edit(int id, [Bind("Id,Name,Address,Establishme
             {
                 try
                 {
-                    _context.Update(parkingZone);
-                    await _context.SaveChangesAsync();
+                    _repository.Update(parkingZone);
+
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/ParkingZoneController.cs]: unnecessary line

File: ParkingZoneApp/Repositories/Repository.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        public void Insert(T entity)
+        {
+            _dbSet.Add(entity);
+            _context.SaveChanges();
+        }
+
+        public void Update(T entity)
+        {
+            _context.Update(entity);    
+            _context.SaveChanges();
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Repositories/Repository.cs:46]: separate the SaveChanges() to the other method called Save()

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

File: ParkingZoneApp/Repositories/Repository.cs (hunk 2 above)
Comment (iamurtazo) [reply ParkingZoneApp/Repositories/Repository.cs:46]: Should i do it only for this Repository.cs class or declare the method in IRepository.cs interface then implement it in the Repository.cs class?

Comment (abdullokhtolibjonov) [reply ParkingZoneApp/Repositories/Repository.cs:46]: oh you have read it, i will say it in meeting

File: ParkingZoneApp/Areas/Admin/ParkingZoneController.cs [hunk 3]
```diff
@@ -7,35 +7,35 @@
...
-            _context = context;
+            _repository = repository;
         }
 
         // GET: Admin/ParkingZone
-        public async Task<IActionResult> Index()
+        public IActionResult Index()
         {
-            return View(await _context.ParkingZones.ToListAsync());
+            return View(_repository.GetAll());
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/ParkingZoneController.cs]: assign to a variable then return

File: ParkingZoneApp/Areas/Admin/ParkingZoneController.cs [hunk 4]
```diff
@@ -138,21 +135,20 @@ public async Task<IActionResult> Delete(int? id)
...
-            var parkingZone = await _context.ParkingZones.FindAsync(id);
+            var parkingZone = _repository.GetById(id);
             if (parkingZone != null)
             {
-                _context.ParkingZones.Remove(parkingZone);
+                _repository.Delete(parkingZone);
             }
 
-            await _context.SaveChangesAsync();
             return RedirectToAction(nameof(Index));
```
Comment (abdullokhtolibjonov) [inline ParkingZoneApp/Areas/Admin/ParkingZoneController.cs]:
```suggestion
            return NotFound();
```

**Review by abdullokhtolibjonov — CHANGES_REQUESTED** [review]

**Review by abdullokhtolibjonov — APPROVED** [review]

**Pull request merged**